parcel.go

```diff
@@ -13,48 +13,88 @@ func NewParcelStore(db *sql.DB) ParcelStore {
 }
 
 func (s ParcelStore) Add(p Parcel) (int, error) {
-	// реализуйте добавление строки в таблицу parcel, используйте данные из переменной p
-
-	// верните идентификатор последней добавленной записи
-	return 0, nil
+	res, err := s.db.Exec("INSERT INTO parcel (client, status, address, created_at) VALUES (:client, :status, :address, :created_at)",
+        sql.Named("client", p.Client),
+    	sql.Named("status", p.Status),
+    	sql.Named("address", p.Address),
+    	sql.Named("created_at", p.CreatedAt))
+	if err != nil {
+        return 0, err
+    }
+
+    id, err := res.LastInsertId()
+    if err != nil {
+        return 0, err
+    }
+
+	return int(id), nil
 }
 
 func (s ParcelStore) Get(number int) (Parcel, error) {
-	// реализуйте чтение строки по заданному number
-	// здесь из таблицы должна вернуться только одна строка
+	row := s.db.QueryRow("SELECT number, client, status, address, created_at FROM parcel WHERE number = :n", 
+		sql.Named("n", number))
 
-	// заполните объект Parcel данными из таблицы
 	p := Parcel{}
+	err := row.Scan(&p.Number, &p.Client, &p.Status, &p.Address, &p.CreatedAt)
+	if err != nil {
+        return p, err
+    }
 
 	return p, nil
 }
 
 func (s ParcelStore) GetByClient(client int) ([]Parcel, error) {
-	// реализуйте чтение строк из таблицы parcel по заданному client
-	// здесь из таблицы может вернуться несколько строк
-
-	// заполните срез Parcel данными из таблицы
 	var res []Parcel
 
+	rows, err := s.db.Query("SELECT number, client, status, address, created_at FROM parcel WHERE client = :n", 
+		sql.Named("n", client))
+    if err != nil {
+        return res, err
+    }
+    defer rows.Close()
+
+    for rows.Next() {
+        p := Parcel{}
+
+        err := rows.Scan(&p.Number, &p.Client, &p.Status, &p.Address, &p.CreatedAt)
+        if err != nil {
+            return res, err
+        }
+
+        res = append(res, p)
+    }
 	return res, nil
 }
 
 func (s ParcelStore) SetStatus(number int, status string) error {
-	// реализуйте обновление статуса в таблице parcel
+	_, err := s.db.Exec("UPDATE parcel SET status = :status WHERE number = :n",
+		sql.Named("status", status),
+		sql.Named("n", number))
+	if err != nil {
+        return err
+    }
 
 	return nil
 }
 
 func (s ParcelStore) SetAddress(number int, address string) error {
-	// реализуйте обновление адреса в таблице parcel
-	// менять адрес можно только если значение статуса registered
+	_, err := s.db.Exec("UPDATE parcel SET address = :address WHERE number = :n AND status = :s",
+		sql.Named("address", address),
+		sql.Named("n", number),
+		sql.Named("s", "registered"))
```

Comment:
Лучше использовать константы
```suggestion
		sql.Named("s", ParcelStatusRegistered))
```